### ISSUE TITLE
why exit query masking?

### ISSUE THREAD

**Issue by ShaoQiBNU** [Nealcly/LAN#13]
In lstm_attention.py , there exit the following code:

\# Query Masking
query_masks = torch.sign(torch.abs(torch.sum(queries, dim=-1)))  # (N, T_q)
query_masks = query_masks.repeat(self.num_heads, 1)  # (h*N, T_q)
query_masks = torch.unsqueeze(query_masks, 2).repeat(1, 1, keys.size()[1])  # (h*N, T_q, T_k)
outputs = outputs * query_masks

The query_masks seem to be a tensor which contain 0 and 1, what is the effect of it and why does it exit ?

**Comment by Nealcly** [comment]
We need to mask the sequence padded for batch.

**Comment by ShaoQiBNU** [comment]
Thank you! I got it.
